blockchains/erc20/erc20_worker.js

```diff
@@ -19,15 +20,24 @@ class ERC20Worker extends BaseWorker {
 
     logger.info(`Connecting to Ethereum node ${constants.NODE_URL}`);
     this.web3 = new Web3(new Web3.providers.HttpProvider(constants.NODE_URL));
+    if (constants.EXPORT_BLOCKS_LIST === '1') {
+      this.blocksListExporter = new BlocksListExporter();
+    }
   }
 
   async init(exporter) {
     this.lastConfirmedBlock = await this.web3.eth.getBlockNumber() - constants.CONFIRMATIONS;
     setGlobalTimestampManager(exporter);
+    if (constants.EXPORT_BLOCKS_LIST === '1') {
+      await this.blocksListExporter.initInWorker();
+    }
   }
 
   async work() {
-    const result = await nextIntervalCalculator(this);
+    const result = (constants.EXPORT_BLOCKS_LIST === '1') ?
+      this.blocksListExporter.calculateArrayPosition() :
```

Comment:
After 1+ h in this PR I start to realize that this feature is meant to read the block numbers to extract ONLY from S3 and the data is still read from the Node, what's the purpose of this? That block list could easily have been provided through the deploy?

Reply:
You can't really give it a list of 5mil blocks through the deploy though. I've initially thought the same, but how would you go onto specifying so much data in the deploy?

Reply:
What's the use case here? Give 5mil random blocks? You always extract a range.

Reply:
We need a way to extract just a portion of the blocks, not all of them. In the case of a BEP20 rerun, you'll need to extract info for 30mil blocks, when sometimes all you need is just 5mil of them. Sometimes, it's not a range but rather they're spread out. Or maybe I don't get the question?

Reply:
> you'll need to extract info for 30mil blocks, when sometimes all you need is just 5mil of them.

You can easily provide the range start and range end through the deploy. If you have 10+ ranges, you may choose to use a ConfigMap for readability.

> it's not a range but rather they're spread out

I can't imagine how this has happened, every 3rd block is wrong or what? Can't you break this up into multiple ranges?

Reply:
If you have a huge number of randomly broken blocks, something is generally wrong with the exporter. I don't this this is a use case we want to have as part of our playbook.

Reply:
> If you have a huge number of randomly broken blocks, something is generally wrong with the exporter.
As is the current case, you don't really know that until it happens. And sometimes you don't have the resources to get a full range of blocks. I'm not sure why this wouldn't be beneficial, as it minimises the data that's going to be extracted
>  If you have 10+ ranges, you may choose to use a ConfigMap for readability.
I'm not sure how you'd do that with 5mil blocks, if there are a lot of gaps and you'd have to manually consider these.

Reply:
> As is the current case, you don't really know that until it happens. 

What does that mean? If you have an exporter that randomly fails blocks, that's pretty bad, just re-export everything. If you are about to say that this is slow - you may have worked on implementing parallel export, this would have been much better use of time. We own the Full Node, we may hit it hard.

> And sometimes you don't have the resources to get a full range of blocks

How come?

> I'm not sure why this wouldn't be beneficial, as it minimises the data that's going to be extracted

I'm not against specifying a range of blocks, this makes tons of sense. The problem is why do you use a storage provider service (S3) to specify those blocks?

> I'm not sure how you'd do that with 5mil blocks, if there are a lot of gaps and you'd have to manually consider these.

Reexport everything.